platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -665,6 +692,26 @@ - (id)mgl_jsonExpressionObject {
                     [expressionObject addObject:obj.mgl_jsonExpressionObject];
                 }];
                 [expressionObject addObject:self.operand.mgl_jsonExpressionObject];
+                return expressionObject;
+            } else if ([function isEqualToString:@"mgl_matchWithOptions:default:"]) {
+                NSMutableArray *expressionObject = [NSMutableArray arrayWithObjects:@"match", self.operand.mgl_jsonExpressionObject, nil];
+                NSArray *optionsArray = self.arguments[0].constantValue;
+                for (NSDictionary<id, NSExpression*> *options in optionsArray) {
+                    for (NSNumber *key in options.allKeys) {
+                        [expressionObject addObject:key];
+                        [expressionObject addObject:[options[key] mgl_jsonExpressionObject]];
+                    }
+                }
+                
+                [expressionObject addObject:[self.arguments[1] mgl_jsonExpressionObject]];
+                return expressionObject;
+            } else if ([function isEqualToString:@"mgl_coalesce"]) {
+                NSMutableArray *expressionObject = [NSMutableArray arrayWithObjects:@"coalesce", nil];
+                
+                for (NSExpression *expression in self.operand.constantValue) {
```

Comment:
It’s a bit awkward to say `FUNCTION({1, 2, 3}, 'mgl_coalesce')`, since people normally think about coalescing an array, not telling the array to coelesce itself. Fortunately, built-in expressions don’t have operands (at least not publicly), so #11472 will implement an `mgl_match:` function that takes a single argument. But this implementation should stay in case the aftermarket expression mechanism fails in the future.